File: frontend/src/types/project.ts

export interface Project {
  id: string;
  name: string;
  description?: string;
  slug: string;
  visibility: 'private' | 'organization' | 'public';
  settings: ProjectSettings;
  tags?: string[];
  avatar?: string;
  organizationId: string;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
  isActive: boolean;
}

export interface ProjectSettings {
  defaultEnvironment: 'development' | 'staging' | 'production';
  experimentRetention: number; // days
  notifications: {
    deployments: boolean;
    experiments: boolean;
    alerts: boolean;
    teamChanges: boolean;
  };
  integrations: {
    gitRepository?: string;
    slackChannel?: string;
  };
  advanced: {
    resourceLimits?: {
      cpu: number;
      memory: number;
    };
    customDomain?: string;
  };
}

export interface ProjectMember {
  id: string;
  userId: string;
  projectId: string;
  role: 'owner' | 'contributor' | 'viewer';
  user: {
    id: string;
    email: string;
    fullName: string;
    avatar?: string;
  };
  addedBy: string;
  createdAt: string;
}

export interface CreateProjectRequest {
  name: string;
  description?: string;
  visibility: 'private' | 'organization' | 'public';
  tags?: string[];
  settings: ProjectSettings;
  members?: {
    userId: string;
    role: 'contributor' | 'viewer';
  }[];
}

export interface UpdateProjectRequest extends Partial<CreateProjectRequest> {
  id: string;
}

export interface ProjectFormData {
  // Basic Information
  name: string;
  description: string;
  visibility: 'private' | 'organization' | 'public';
  tags: string[];
  avatar?: File;

  // Team Configuration
  members: {
    email: string;
    role: 'contributor' | 'viewer';
  }[];
  bulkEmails: string;

  // Settings
  defaultEnvironment: 'development' | 'staging' | 'production';
  experimentRetention: number;
  notifications: {
    deployments: boolean;
    experiments: boolean;
    alerts: boolean;
    teamChanges: boolean;
  };
  gitRepository: string;
  slackChannel: string;
  resourceLimits: {
    cpu: number;
    memory: number;
  };
  customDomain: string;
}

File: backend/app/schemas/project.py
from pydantic import BaseModel, Field
from typing import Optional, List, Literal
from datetime import datetime
import uuid


class ProjectBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = Field(None, max_length=2000)
    visibility: Literal["private", "organization", "public"] = "private"
    settings: dict = Field(default_factory=dict)


class ProjectCreate(ProjectBase):
    organization_id: uuid.UUID


class ProjectUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = Field(None, max_length=2000)
    visibility: Optional[Literal["private", "organization", "public"]] = None
    settings: Optional[dict] = None


class ProjectResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: Optional[str]
    slug: str
    visibility: str
    settings: dict
    organization_id: uuid.UUID
    created_by: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ProjectListResponse(BaseModel):
    projects: List[ProjectResponse]
    total: int
    page: int
    size: int


class ProjectMemberResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    project_id: uuid.UUID
    role: Literal["owner", "contributor", "viewer"]
    created_at: datetime

    class Config:
        from_attributes = True


